chains/atomic/mocks_generate_test.go

```diff
@@ -0,0 +1,6 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package atomic
+
+//go:generate go run go.uber.org/mock/mockgen@v0.5 -package=${GOPACKAGE}mock -destination=${GOPACKAGE}mock/shared_memory.go -mock_names=SharedMemory=SharedMemory . SharedMemory
```

Comment:
Is it not possible to get:
```suggestion
//go:generate go run go.uber.org/mock/mockgen -package=${GOPACKAGE}mock -destination=${GOPACKAGE}mock/shared_memory.go -mock_names=SharedMemory=SharedMemory . SharedMemory
```
to work correctly somehow?

I know support for [this is explicitly added in go1.24](https://go.dev/doc/modules/managing-dependencies#tools) but they seem to suggest that the current norm for doing this is by including a `tools.go` file with the required dependencies.
> This removes the need for the previous workaround of adding tools as blank imports to a file conventionally named “tools.go”.

I feel like changing the mockgen version shouldn't require changing a ton of files explicitly.

Reply:
If we don't specify the version, it requires to run `go get go.uber.org/mock/mockgen@v0.5` to add the dependency `golang.org/x/mod v0.18.0 // indirect` in `go.mod`, which gets removed by a `go mod tidy`. So indeed the solution is to add a `tools.go` see commit 0ae62d225ec10318c11370e7669063478573d1f3

I think it's nicer, although feel free to suggest a better location for tools.go since the root of the project might not be the best place.

Reply:
Created issue https://github.com/ava-labs/avalanchego/issues/3633 to track this and use the future `tool` directive.